packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -593,43 +594,23 @@ export class GarbageCollector implements IGarbageCollector {
 	): { tombstoneReadyNodeIds: Set<string>; sweepReadyNodeIds: Set<string> } {
 		// 1. Marks all referenced nodes by clearing their unreferenced tracker, if any.
 		for (const nodeId of allReferencedNodeIds) {
-			const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
-			if (nodeStateTracker !== undefined) {
-				// Stop tracking so as to clear out any running timers.
-				nodeStateTracker.stopTracking();
-				// Delete the node as we don't need to track it any more.
-				this.unreferencedNodesState.delete(nodeId);
-			}
+			this.unreferencedNodesState.delete(nodeId);
 		}
 
-		// 2. Mark unreferenced nodes in this run by starting unreferenced tracking for them.
+		// 2. Mark unreferenced nodes in this run by starting or updating unreferenced tracking for them.
 		const tombstoneReadyNodeIds: Set<string> = new Set();
 		const sweepReadyNodeIds: Set<string> = new Set();
 		for (const nodeId of gcResult.deletedNodeIds) {
-			const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
-			if (nodeStateTracker === undefined) {
-				this.unreferencedNodesState.set(
-					nodeId,
-					new UnreferencedStateTracker(
-						currentReferenceTimestampMs,
-						this.configs.inactiveTimeoutMs,
-						currentReferenceTimestampMs,
-						this.configs.tombstoneTimeoutMs,
-						this.configs.sweepGracePeriodMs,
-					),
-				);
-			} else {
-				// If a node was already unreferenced, update its tracking information. Since the current reference time
-				// is from the ops seen, this will ensure that we keep updating unreferenced state as time moves forward.
-				nodeStateTracker.updateTracking(currentReferenceTimestampMs);
-
-				// If a node is tombstone or sweep-ready, store it so it can be returned.
-				if (nodeStateTracker.state === UnreferencedState.TombstoneReady) {
-					tombstoneReadyNodeIds.add(nodeId);
-				}
-				if (nodeStateTracker.state === UnreferencedState.SweepReady) {
-					sweepReadyNodeIds.add(nodeId);
-				}
+			const state = this.unreferencedNodesState.startOrUpdateTracking(
```

Comment:
I think this helper `startOrUpdateTracking` is not needed. It just hides the implementation and the fact that it returns the state is not obvious. Using the underlying get and then set makes it clear what's happening here.

Reply:
Also, the part about updating the tracking is no longer needed. The unreferenced state is updated on every connection. For summarizer clients, this will happen when they transition to "write" mode and will have a recent timestamp. Doing this on every GC run is unncessary and expensive.
It is unrelated to this PR of course.

Reply:
K I'll revert this.  And I put a note in AB#1651 about the unnecessary reboot of the timers.